src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -0,0 +1,296 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {Platform} from '@angular/cdk/platform';
+import {
+  AfterViewInit,
+  Attribute,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Inject,
+  Input,
+  NgZone,
+  OnDestroy,
+  Optional,
+  Output,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+import {
+  MAT_CHECKBOX_CLICK_ACTION,
+  MatCheckboxClickAction,
+  RippleRenderer,
+  RippleTarget,
+  ThemePalette
+} from '@angular/material';
+import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
+import {MDCCheckboxAdapter, MDCCheckboxFoundation} from '@material/checkbox';
+
+let nextUniqueId = 0;
+
+export const MAT_MDC_CHECKBOX_CONTROL_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => MatCheckbox),
+  multi: true
+};
+
+/** Change event object emitted by MatCheckbox. */
+export class MatCheckboxChange {
+  /** The source MatCheckbox of the event. */
+  source: MatCheckbox;
+  /** The new `checked` value of the checkbox. */
+  checked: boolean;
+}
+
+@Component({
+  moduleId: module.id,
+  selector: 'mat-checkbox',
+  templateUrl: 'checkbox.html',
+  styleUrls: ['checkbox.css'],
+  host: {
+    'class': 'mat-mdc-checkbox',
+    '[attr.tabindex]': 'null',
+    '[class.mat-primary]': 'color == "primary"',
+    '[class.mat-accent]': 'color == "accent"',
+    '[class.mat-warn]': 'color == "warn"',
+    '[class._mat-animation-noopable]': `_animationMode === 'NoopAnimations'`,
+    '[id]': 'id',
+    '(animationend)': '_checkboxFoundation.handleAnimationEnd()',
```

Comment:
I'm not sure... it might call back into our code via the adapter. Looking at this again I wonder if I should try to remove this listener all together. Stuff like this won't be reliable if we disable animations.